src/main/java/setup/BaseTest.java

```diff
@@ -1,67 +1,166 @@
 package setup;
 
 import io.appium.java_client.AppiumDriver;
+import io.appium.java_client.AppiumFluentWait;
+import java.net.URLEncoder;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
 import org.openqa.selenium.remote.DesiredCapabilities;
-import org.testng.annotations.*;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Optional;
+import org.testng.annotations.AfterClass;
+import pageObjects.NativePages.BudgetActivityPage;
+import pageObjects.NativePages.LogInPage;
+import pageObjects.NativePages.RegisterPage;
 import pageObjects.PageObject;
-
 import java.io.File;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.concurrent.TimeUnit;
+import pageObjects.WebPages.GooglePage;
 
 public class BaseTest implements IDriver {
 
     private static AppiumDriver appiumDriver; // singleton
     IPageObject po;
 
+    LogInPage logInPage;
+    RegisterPage registerPage;
+
+    AppiumFluentWait webDriverWait;
+    BudgetActivityPage budgetActivityPage;
+
+    static final int waitTime = 10;
+    static final String isPageReadyScript = "return document.readyState";
+    static final String isPageReadyScriptResult = "complete";
+
     @Override
-    public AppiumDriver getDriver() { return appiumDriver; }
+    public AppiumDriver getDriver() {
+        return appiumDriver;
+    }
 
     public IPageObject getPo() {
         return po;
     }
 
-    @Parameters({"platformName","appType","deviceName","browserName","app"})
-    @BeforeSuite(alwaysRun = true)
-    public void setUp(String platformName, String appType, String deviceName, @Optional("") String browserName, @Optional("") String app) throws Exception {
-        System.out.println("Before: app type - "+appType);
-        setAppiumDriver(platformName, deviceName, browserName, app);
+    @Parameters({"platformName", "appType", "deviceName", "udid", "browserName", "app", "appPackage", "appActivity",
+        "bundleId"})
+    @BeforeClass(alwaysRun = true) //changed for running two tests in one suite
+    public void setUp(String platformName,
+                      String appType,
+                      @Optional("") String deviceName,
+                      @Optional("") String udid,
+                      @Optional("") String browserName,
+                      @Optional("") String app,
+                      @Optional("") String appPackage,
+                      @Optional("") String appActivity,
+                      @Optional("") String bundleId) throws Exception {
```

Comment:
It's better not to throw exception out of app (where it won't be processed)

Reply:
Try - catch added.